src/main/java/com/authlete/common/util/security/AESCipher.java

```diff
@@ -0,0 +1,581 @@
+import org.apache.commons.codec.BinaryEncoder;
+import org.apache.commons.codec.binary.Base64;
+
+/**
+ * Cipher using {@code "AES/CBC/PKCS5Padding"}.
+ *
+ * <pre style="background-color: #EEEEEE; margin: 2em; border: 1px solid black; padding: 0.5em;">
+ * <span style="color: darkgreen;">// Create a cipher with a secret key.</span>
+ * AESCipher cipher = new {@link #AESCipher()}.{@link #setKey(String, String) setKey}(<span style="color: darkred;">"secret key"</span>, <span style="color: darkred;">"initial vector"</span>);
+ *
+ * <span style="color: darkgreen;">// Encryption &amp; decryption.
+ * // 'plaintext' and 'decrypted' have the same value.</span>
+ * String plaintext = <span style="color: darkred;">"plain text"</span>;
+ * String encrypted = cipher.{@link #encrypt(String) encrypt(plaintext)};
+ * String decrypted = cipher.{@link #decrypt(String) decrypt(encrypted)};
+ *
+ * <span style="color: darkgreen;">// In the above example, 'encrypted' is encoded by Base64 (default).
+ * // If you want to change the format, use {@code setCoder} method.
+ * // For example, to change the format to hexadecimal:</span>
+ * Hex hex = new org.apache.commons.codec.binary.Hex();
+ * cipher.setCoder(hex);
+ *
+ * <span style="color: darkgreen;">// Binary representation (only "0"s and "1"s) also can be used.</span>
+ * BinaryCodec binary = new org.apache.commons.codec.BinaryCodec();
+ * cipher.setCoder(binary);
+ *
+ * <span style="color: darkgreen;">// Coder can be specified as a constructor parameter.</span>
+ * cipher = new AESCipher(hex);
+ *
+ * <span style="color: darkgreen;">// If you want, an encoder and a decoder can be set separately.</span>
+ * cipher.{@link #setEncoder(BinaryEncoder) setEncoder(hex)};
+ * cipher.{@link #setDecoder(BinaryDecoder) setDecoder(hex)};
+ * </pre>
+ *
+ * <pre style="background-color: #EEEEEE; margin: 2em; border: 1px solid black; padding: 0.5em;">
+ * <span style="color: darkgreen;">// Another example which performs encryption without initial vector.</span>
+ * String secretkey = <span style="color: darkred;">"secret key"</span>;
+ * String plaintext = <span style="color: darkred;">"plain text"</span>;
+ *
+ * <span style="color: darkgreen;">// Create and set up without initial vector.</span>
+ * AESCipher cipher = new AESCipher().setKey(secretkey);
+ *
+ * <span style="color: darkgreen;">// Encrypt.</span>
+ * String encrypted = cipher.encrypt(plaintext);
+ *
+ * <span style="color: darkgreen;">// Get the auto-generated initial vector.</span>
+ * byte[] iv = cipher.getCipher().getIV();
+ *
+ * <span style="color: darkgreen;">// Decryption requires initial vector.</span>
+ * cipher.setKey(secretkey, iv);
+ *
+ * <span style="color: darkgreen;">// Decrypt.</span>
+ * String decrypted = cipher.decrypt(encrypted);
+ * </pre>
+ *
+ * @author Takahiko Kawasaki
+ */
+public class AESCipher extends CodecCipher
+{
+    /**
+     * The default key size. The value is 16, meaning 128 bits.
+     *
+     * @since 1.4
```

Comment:
I think we should remove this "since xxx" copied from the original code. Same for other properties and methods.